internal/declcfg/helpers_test.go

```diff
@@ -40,14 +40,14 @@ func buildValidDeclarativeConfig(includeUnrecognized bool) DeclarativeConfig {
 			{Schema: "custom.1", Blob: json.RawMessage(`{"schema": "custom.1"}`)},
 			{Schema: "custom.2", Blob: json.RawMessage(`{"schema": "custom.2"}`)},
 			{Schema: "custom.3", Package: "anakin", Blob: json.RawMessage(`{
-				"schema": "custom.3",
+				"myField": "foobar",
 				"package": "anakin",
-				"myField": "foobar"
+				"schema": "custom.3"
 			}`)},
 			{Schema: "custom.3", Package: "boba-fett", Blob: json.RawMessage(`{
-				"schema": "custom.3",
+				"myField": "foobar",
 				"package": "boba-fett",
-				"myField": "foobar"
+				"schema": "custom.3"
```

Comment:
No, this order changed because the yaml encoder does not honor struct field order. It looks like it may just be alphanumeric order.

Reply:
Well that seems like a bug in the encoder, no?

Reply:
YAML _is_ field order agnostic. This change is just necessary because the roundtrip loses the original ordering. There's no bug here unless we want declarative config blobs to roundtrip map field orders. 

I think this happens because the underlying library is sigs.k8s.io/yaml, which encodes `value -> JSON -> YAML` and decodes `YAML -> JSON -> value`.

I think this is due to the JSON <-> YAML step, which uses what amounts to a `map[string]interface{}` for JSON/YAML maps, and Golang maps are not ordered.